lib/mix/tasks/inertia.install.ex

```diff
@@ -0,0 +1,332 @@
+  @doc false
+  def setup_router(igniter) do
+    Igniter.Libs.Phoenix.append_to_pipeline(igniter, :browser, "plug Inertia.Plug")
+  end
+
+  @doc false
+  def add_inertia_config(igniter) do
+    # Get endpoint module name based on app name
+    {igniter, endpoint_module} = Igniter.Libs.Phoenix.select_endpoint(igniter)
+
+    # Determine configuration based on options
+    camelize_props = igniter.args.options[:camelize_props] || false
+    history_encryption = igniter.args.options[:history_encrypt] || false
+
+    config_options = [
+      endpoint: endpoint_module
+    ]
+
+    # Add camelize_props config if specified
+    config_options =
+      if camelize_props do
+        Keyword.put(config_options, :camelize_props, true)
+      else
+        config_options
+      end
+
+    # Add history encryption config if specified
+    config_options =
+      if history_encryption do
+        Keyword.put(config_options, :history, encrypt: true)
+      else
+        config_options
+      end
+
+    # Add the configuration to config.exs
+    Enum.reduce(config_options, igniter, fn {key, value}, igniter ->
+      Igniter.Project.Config.configure(
+        igniter,
+        "config.exs",
+        :inertia,
+        [key],
+        value
+      )
+    end)
+  end
+
+  @doc false
+  def update_root_layout(igniter) do
+    file_path =
+      Path.join([
+        "lib",
+        web_dir(igniter),
+        "components",
+        "layouts",
+        "root.html.heex"
+      ])
+
+    content = inertia_root_html()
+
+    Igniter.create_new_file(igniter, file_path, content, on_exists: :overwrite)
+  end
+
+  defp web_dir(igniter) do
+    igniter
+    |> Igniter.Libs.Phoenix.web_module()
+    |> inspect()
+    |> Macro.underscore()
+  end
+
+  defp inertia_root_html() do
+    """
+    <!DOCTYPE html>
+    <html lang="en">
+      <head>
+        <meta charset="utf-8" />
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
+        <meta name="csrf-token" content={get_csrf_token()} />
+        <.inertia_title><%= assigns[:page_title] %></.inertia_title>
+        <.inertia_head content={@inertia_head} />
+        <link phx-track-static rel="stylesheet" href={~p"/assets/css/app.css"} />
+        <script type="module" defer phx-track-static src={~p"/assets/app.js"} />
+      </head>
+      <body>
+        {@inner_content}
+      </body>
+    </html>
+    """
+  end
+
+  @doc false
+  def update_esbuild_config(igniter) do
+    igniter
+    |> Igniter.Project.Config.configure(
+      "config.exs",
+      :esbuild,
+      [:version],
+      "0.21.5"
+    )
+    |> Igniter.Project.Config.configure(
+      "config.exs",
+      :esbuild,
+      [Igniter.Project.Application.app_name(igniter)],
+      {:code,
+       Sourceror.parse_string!("""
+       [
+        args:
+          ~w(js/app.jsx --bundle --chunk-names=chunks/[name]-[hash] --splitting --format=esm  --target=es2020 --outdir=../priv/static/assets --external:/fonts/* --external:/images/*),
```

Comment:
@derrickreimer would you recommend code splitting be on by default when using the installer, or leave it off?

Reply:
Good question. I think it's worth making it on by default.